R/projectSpaceCooling.R

```diff
@@ -0,0 +1,211 @@
+                                config,
+                                acOwnershipRates,
+                                regPars,
+                                endOfHistory,
+                                lambda,
+                                outliers = NULL) {
+
+  # PARAMETERS -----------------------------------------------------------------
+
+  # current scenario
+  scen <- unique(data$scenario[data$scenario != "history"]) %>%
+    as.character()
+
+  day2sec <- 24 * 3600 # h/d * s/h
+
+  # assumed future relative growth of activity factor between EOH and 2070
+  activityBoost <- config[, "coolingActivityBoost"] %>%
+    as.data.frame()
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  cols <- colnames(data)
+
+  regPars <- setNames(regPars$value, regPars$variable)
+
+
+  ## Regional beta corrections ====
+
+  # full data set
+  modelData <- acOwnershipRates %>%
+    filter(!is.na(.data$value),
+           .data$value != 0) %>%
+    select("region", "period", "value") %>%
+    rename("penetration" = "value") %>%
+
+    # join GDP per capita and CDD data
+    left_join(data %>%
+                filter(.data$variable %in% c("CDD", "gdppop")) %>%
+                select("region", "period", "variable", "value") %>%
+                pivot_wider(names_from = "variable", values_from = "value"),
+              by = c("region", "period"))
+
+
+  # extract global fit parameters
+  alpha <- regPars[["alpha"]]
+  beta  <- regPars[["beta"]]
+  gamma <- regPars[["gamma"]]
+  delta <- regPars[["delta"]]
+
+
+  betaReg <- modelData %>%
+    # match last historical data point
+    group_by(across(all_of(c("region")))) %>%
+    slice_max(order_by = .data$period, n = 1, with_ties = FALSE) %>%
+    ungroup() %>%
+    mutate(betaReg = (alpha - log(1 / .data$penetration - 1)) / (.data$gdppop^delta * .data$CDD^gamma),
+           betaReg = ifelse(.data$betaReg < 0, NA, .data$betaReg)) %>%
+
+    # fill missing periods with global beta
+    right_join(data %>%
+                 select("region") %>%
+                 unique(),
+               by = "region") %>%
+    mutate(betaReg = ifelse((is.na(.data$betaReg) | .data$betaReg < 0),
+                            beta,
+                            betaReg),
+           # use global beta for outlier regions
+           betaReg = ifelse(.data$region %in% outliers, beta, betaReg)) %>%
+    select("region", "betaReg")
+
+
+  ## Projections ====
+
+  # project penetration rate
+  projectionData <- data %>%
+    filter(.data$variable %in% c("CDD", "gdppop", "space_cooling", "buildings", "uvalue")) %>%
+    select("region", "period", "variable", "value") %>%
+    pivot_wider(names_from = "variable", values_from = "value") %>%
+    left_join(betaReg, by = "region") %>%
+    mutate(acPenetration = 1 / (1 + exp(alpha - .data$betaReg * .data$gdppop^delta * .data$CDD^gamma))) %>%
+
+    # calculate unscaled demand to determine regional activity factors
```

Comment:
Minor addition to understand the conversion
```suggestion
    # calculate unscaled demand (EJ/yr) to determine regional activity factors
```